From 66f97bc0d5a94ace1befb9852365f382e1accee4 Mon Sep 17 00:00:00 2001
From: JeevakaPrabu <jeevaka.badrappan@intel.com>
Date: Mon, 10 Feb 2025 01:23:35 -0500
Subject: [PATCH] Add Android.bp file to compile with soong

This build will enable wlan build for A16

Tests Done: Wifi is working

Tracked-On: NA
Signed-off-by: JeevakaPrabu <jeevaka.badrappan@intel.com>
---
 Android.mk                      |  5 ---
 wpa_supplicant_8_lib/Android.bp | 65 ++++++++++++++++++++++++++++++++
 wpa_supplicant_8_lib/Android.mk | 66 ---------------------------------
 3 files changed, 65 insertions(+), 71 deletions(-)
 delete mode 100644 Android.mk
 create mode 100644 wpa_supplicant_8_lib/Android.bp
 delete mode 100644 wpa_supplicant_8_lib/Android.mk

diff --git a/Android.mk b/Android.mk
deleted file mode 100644
index 59247e9..0000000
--- a/Android.mk
+++ /dev/null
@@ -1,5 +0,0 @@
-LOCAL_PATH := $(call my-dir)
-
-include $(CLEAR_VARS)
-
-include $(LOCAL_PATH)/wpa_supplicant_8_lib/Android.mk
diff --git a/wpa_supplicant_8_lib/Android.bp b/wpa_supplicant_8_lib/Android.bp
new file mode 100644
index 0000000..55faee6
--- /dev/null
+++ b/wpa_supplicant_8_lib/Android.bp
@@ -0,0 +1,65 @@
+//
+// Copyright (C) 2008 The Android Open Source Project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package {
+    // See: http://go/android-license-faq
+    default_applicable_licenses: [
+        "intel_wifi_wpa_supplicant_8_lib_license",
+    ],
+}
+
+license {
+    name: "intel_wifi_wpa_supplicant_8_lib_license",
+    visibility: [":__subpackages__"],
+    license_kinds: [
+        "SPDX-license-identifier-Apache-2.0",
+    ],
+}
+
+cc_library_static {
+    name: "lib_driver_cmd_intc",
+    shared_libs: [
+        "libc",
+        "libcutils",
+        "liblog",
+    ],
+    header_libs: [
+        "libcutils_headers",
+        "libhardware_headers",
+        "device_kernel_headers",
+    ],
+    cflags: [
+        "-DCONFIG_ANDROID_LOG",
+        "-Wno-unused-parameter",
+    ],
+    srcs: [
+        "driver_cmd_nl80211.c",
+    ],
+    include_dirs: [
+        "external/libnl/include",
+        "external/wpa_supplicant_8/src",
+        "external/wpa_supplicant_8/src/common",
+        "external/wpa_supplicant_8/src/drivers",
+        "external/wpa_supplicant_8/src/l2_packet",
+        "external/wpa_supplicant_8/src/utils",
+        "external/wpa_supplicant_8/src/wps",
+        "external/wpa_supplicant_8/wpa_supplicant",
+    ],
+    vendor: true,
+    defaults: [
+        "wpa_supplicant_cflags_default",
+    ],
+}
diff --git a/wpa_supplicant_8_lib/Android.mk b/wpa_supplicant_8_lib/Android.mk
deleted file mode 100644
index d79f12b..0000000
--- a/wpa_supplicant_8_lib/Android.mk
+++ /dev/null
@@ -1,66 +0,0 @@
-#
-# Copyright (C) 2014 The Android Open Source Project
-# Copyright (C) 2019 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-LOCAL_PATH := $(call my-dir)
-
-ifeq ($(WPA_SUPPLICANT_VERSION),VER_2_1_DEVEL)
-
-ifneq ($(BOARD_WPA_SUPPLICANT_DRIVER),)
-  CONFIG_DRIVER_$(BOARD_WPA_SUPPLICANT_DRIVER) := y
-endif
-
-WPA_SUPPL_DIR = external/wpa_supplicant_8
-WPA_SRC_FILE :=
-
-include $(WPA_SUPPL_DIR)/wpa_supplicant/android.config
-
-
-WPA_SUPPL_DIR_INCLUDE = $(WPA_SUPPL_DIR)/src \
-        $(WPA_SUPPL_DIR)/src/common \
-        $(WPA_SUPPL_DIR)/src/drivers \
-        $(WPA_SUPPL_DIR)/src/l2_packet \
-        $(WPA_SUPPL_DIR)/src/utils \
-        $(WPA_SUPPL_DIR)/src/wps \
-        $(WPA_SUPPL_DIR)/wpa_supplicant
-
-ifdef CONFIG_DRIVER_NL80211
-WPA_SUPPL_DIR_INCLUDE += external/libnl/include
-WPA_SRC_FILE += driver_cmd_nl80211.c
-endif
-
-ifdef CONFIG_ANDROID_LOG
-L_CFLAGS += -DCONFIG_ANDROID_LOG
-endif
-L_CFLAGS += -fno-delete-null-pointer-checks \
-            -fwrapv
-
-########################
-
-include $(CLEAR_VARS)
-LOCAL_MODULE := lib_driver_cmd_intc
-LOCAL_PROPRIETARY_MODULE := true
-LOCAL_SHARED_LIBRARIES := libc libcutils libnl libnetutils
-LOCAL_HEADER_LIBRARIES += libutils_headers libhardware_headers
-LOCAL_CFLAGS := $(L_CFLAGS)
-LOCAL_SRC_FILES := $(WPA_SRC_FILE)
-LOCAL_C_INCLUDES := $(WPA_SUPPL_DIR_INCLUDE) \
-	$(LOCAL_PATH)
-LOCAL_SANITIZE := cfi
-include $(BUILD_STATIC_LIBRARY)
-
-########################
-
-endif
-- 
2.34.1

